instrumentation/micrometer/micrometer-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/OpenTelemetryFunctionTimer.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.micrometer.v1_5;
+
+import static io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.Bridging.description;
+import static io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.Bridging.tagsAsAttributes;
+
+import io.micrometer.core.instrument.FunctionTimer;
+import io.micrometer.core.instrument.Measurement;
+import io.micrometer.core.instrument.Statistic;
+import io.micrometer.core.instrument.util.MeterEquivalence;
+import io.micrometer.core.instrument.util.TimeUtils;
+import io.opentelemetry.api.common.Attributes;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToLongFunction;
+
+@SuppressWarnings("HashCodeToString")
+final class OpenTelemetryFunctionTimer<T> implements FunctionTimer, RemovableMeter {
+
+  private final Id id;
+  private final Attributes attributes;
+  private final AsyncInstrumentRegistry asyncInstrumentRegistry;
+  private final String countMeterName;
+  private final String totalTimeMeterName;
+
+  OpenTelemetryFunctionTimer(
+      Id id,
+      T obj,
+      ToLongFunction<T> countFunction,
+      ToDoubleFunction<T> totalTimeFunction,
+      TimeUnit totalTimeFunctionUnit,
+      AsyncInstrumentRegistry asyncInstrumentRegistry) {
+    this.id = id;
+    this.attributes = tagsAsAttributes(id);
+    this.asyncInstrumentRegistry = asyncInstrumentRegistry;
+
+    countMeterName = id.getName() + "." + Statistic.COUNT.getTagValueRepresentation();
+    totalTimeMeterName = id.getName() + "." + Statistic.TOTAL_TIME.getTagValueRepresentation();
+
+    asyncInstrumentRegistry.buildLongCounter(
+        countMeterName, description(id), /* baseUnit = */ "1", attributes, obj, countFunction);
+
+    asyncInstrumentRegistry.buildDoubleCounter(
+        totalTimeMeterName,
+        description(id),
+        /* baseUnit = */ "ms",
+        attributes,
+        obj,
+        new ConvertToMillisDecorator<>(totalTimeFunction, totalTimeFunctionUnit));
```

Comment:
Consider replacing with a one liner:
```suggestion
    asyncInstrumentRegistry.buildDoubleCounter(
        totalTimeMeterName,
        description(id),
        /* baseUnit = */ "ms",
        attributes,
        obj,
        value -> TimeUtils.convert(totalTimeFunction.applyAsDouble(value), totalTimeFunctionUnit, TimeUnit.MILLISECONDS));
```

Reply:
Good point, will do.